rotkehlchen/api/rest.py

```diff
@@ -5420,3 +5421,29 @@ def prepare_native_transfer(
             return wrap_in_fail_result(str(e), status_code=HTTPStatus.BAD_REQUEST)
 
         return _wrap_in_ok_result(result=payload)
+
+    @async_api_call()
+    def fetch_token_balance_for_address(
+            self,
+            address: ChecksumEvmAddress,
+            evm_chain: EVM_CHAIN_IDS_WITH_TRANSACTIONS_TYPE,
+            asset: 'CryptoAsset',
```

Comment:
mmmm the chain is needed if you want to query the native token but if the query is for a token should you check that the chains match? or is it an overkill?

Reply:
i don't think it's needed since nothing gets stored in the database. the call will simply fail.